### PR TITLE
Correctly show unregister-button with seperate deadlines

### PR DESCRIPTION
**Pull request by itsisak**
# Description

Previously, the registration form and button only took the registration deadline into account. So sometimes when you should be able to unregister you couldnt and vise versa.

Also updated the EventDetail info to show the seperate deadlines as 

# Result

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

<table>
    <tr>
        <td width="25%">Description</td>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td>
            Seperate registration and unregistration deadlines
        </td>
        <td> 

Screenshot 2024-10-30 at 20 28 52
        </td>
        <td> 
Screenshot 2024-10-30 at 20 29 25
        </td>
    </tr>
    <tr>
        <td>
            Info on when unregistration closes, like the registration deadline has.
        </td>
        <td> 
Screenshot 2024-10-30 at 20 18 48
        </td>
        <td> 
Screenshot 2024-10-30 at 20 19 43</td>
    </tr>
</table>

# Testing

- [x] I have thoroughly tested my changes.

Tested with a bunch of different registration/unregistration deadline combos. 

---

Resolves ABA-489

### CODE REVIEW

**Review by ollfkaih — APPROVED** [review]
looks good to meh I think sans tests (:

**Review by christiangryt — APPROVED** [review]
LGTM!

**Review by ShaileshS1702 — COMMENTED** [review]
LGTM!

File: app/routes/events/components/EventDetail/infoLists.tsx [hunk 1]
```diff
@@ -105,6 +113,66 @@ export const useDeadlineInfoList = (event?: DetailedEvent) => {
...
+                currentMoment.isBefore(unregistrationCloseTimeMoment)
+                  ? 'Avmelding stenger'
+                  : 'Avmelding stengte'
+              }
+              tooltipContentIcon={
+                <>
+                  Etter avmeldingen stenger er det ikke mulig å melde seg av
+                  arrangementet
+                </>
+              }
```
Comment (ivarnakken) [inline app/routes/events/components/EventDetail/infoLists.tsx:163]: what's happening here? Why is it called [..]Icon when it takes in text?

**Review by ivarnakken — APPROVED** [review]
Very clean code and results! 😍

File: app/routes/events/components/EventDetail/infoLists.tsx (hunk 1 above)
Comment (norbye) [reply app/routes/events/components/EventDetail/infoLists.tsx:163]: Should probably be renamed to `iconTooltipContent` or something as I'm assuming it's the content for the tooltip appearing over the icon

Comment (itsisak) [reply app/routes/events/components/EventDetail/infoLists.tsx]: Agree the `[..]Icon` does not fit there. Renamed to `tooltipContent` as the type indicated, there is only one tooltip in the component so should be good enough name.

**Pull request merged**